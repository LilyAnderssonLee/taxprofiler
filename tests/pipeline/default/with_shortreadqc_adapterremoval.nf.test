nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test"
    tag "pipeline"

    test("Short Read QC Tool - Adapter Removal") {

        when {
            params {
                outdir            = "$outputDir"
                shortread_qc_tool = "adapterremoval"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert new File("$outputDir/bbduk/2612_ERR5766176_B.bbduk.log").exists() },
                { assert new File("$outputDir/bbduk/2612_ERR5766176.bbduk.log").exists() },
                { assert new File("$outputDir/bbduk/2612_ERR5766180.bbduk.log").exists() },
                { assert new File("$outputDir/bbduk/2613_ERR5766181.bbduk.log").exists() },
                { assert snapshot(path("$outputDir/bowtie2/align/").list()).match("bowtie2/align/") },
                { assert snapshot(path("$outputDir/bracken/bracken_db1_combined_reports.txt")).match("bracken") },
                { assert snapshot(path("$outputDir/centrifuge/centrifuge_db3_combined_reports.txt")).match("centrifuge") },
                { assert snapshot(path("$outputDir/diamond/db1/2611_se_db1.diamond.tsv"),
                                path("$outputDir/diamond/db1/2612_se_db1.diamond.tsv"),
                                path("$outputDir/diamond/db1/2613_se_db1.diamond.tsv"),
                                path("$outputDir/diamond/db1/ERR3201952_se_db1.diamond.tsv")).match("diamond") },
                { assert snapshot(path("$outputDir/adapterremoval/").list()).match("adapterremoval") },
                { assert snapshot(path("$outputDir/filtlong/").list()).match("filtlong") },
                { assert snapshot(path("$outputDir/kaiju/kaiju_db5_combined_reports.txt")).match("kaiju") },
                { assert snapshot(path("$outputDir/kraken2/kraken2_db1_combined_reports.txt"),
                                path("$outputDir/kraken2/kraken2_db2_combined_reports.txt")).match("kraken2") },
                { assert new File("$outputDir/krona/centrifuge_db3.html").exists() },
                { assert new File("$outputDir/krona/kaiju_db5.html").exists() },
                { assert new File("$outputDir/krona/kraken2_db1.html").exists() },
                { assert new File("$outputDir/krona/kraken2_db2.html").exists() },
                { assert snapshot(path("$outputDir/metaphlan3/metaphlan3_db4_combined_reports.txt")).match("metaphlan3") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/bbduk.txt"),
                                path("$outputDir/multiqc/multiqc_data/diamond.txt"),
                                path("$outputDir/multiqc/multiqc_data/filtlong.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_bowtie2.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_adapter_removal.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc_1.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_kaiju_species.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_samtools_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/porechop.txt")).match("multiqc") },
                { assert snapshot(UTILS.filterLines("$outputDir/porechop/ERR3201952_ERR3201952.log", -3)).match("porechop") },
                { assert snapshot(path("$outputDir/samtools/stats/").list()).match("samtools/stats/") },
                { assert snapshot(path("$outputDir/taxpasta/").list()).match("taxpasta") },
            )
        }

    }

}
