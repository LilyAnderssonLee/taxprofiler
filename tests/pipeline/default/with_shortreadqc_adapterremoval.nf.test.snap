{
    "filtlong": {
        "content": [
            "ERR3201952_ERR3201952_filtered.log:md5,513650c6b058af60a824f98e6a632e2d"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "metaphlan3": {
        "content": [
            "metaphlan3_db4_combined_reports.txt:md5,ceb88c02370a2c3bccdd3ff4ee7c23a9"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "software_versions": {
        "content": [
            "{ADAPTERREMOVAL_PAIRED={adapterremoval=2.3.2}, ADAPTERREMOVAL_SINGLE={adapterremoval=2.3.2}, BBMAP_BBDUK={bbmap=39.01}, BOWTIE2_ALIGN={bowtie2=2.4.4, pigz=2.6, samtools=1.16.1}, BOWTIE2_BUILD={bowtie2=2.4.4}, BRACKEN_BRACKEN={bracken=2.7}, CENTRIFUGE_CENTRIFUGE={centrifuge=1.0.4}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, DIAMOND_BLASTX={diamond=2.0.15}, FASTQC={fastqc=0.11.9}, FASTQC_PROCESSED={fastqc=0.11.9}, FILTLONG={filtlong=0.2.1}, KAIJU_KAIJU={kaiju=1.8.2}, KAIJU_KAIJU2KRONA={kaiju=1.8.2}, KAIJU_KAIJU2TABLE_COMBINED={kaiju=1.8.2}, KAIJU_KAIJU2TABLE_SINGLE={kaiju=1.8.2}, KRAKEN2_KRAKEN2={kraken2=2.1.2, pigz=2.6}, KRAKENTOOLS_COMBINEKREPORTS_CENTRIFUGE={combine_kreports.py=1.2}, KRAKENTOOLS_COMBINEKREPORTS_KRAKEN={combine_kreports.py=1.2}, KRAKENTOOLS_KREPORT2KRONA={kreport2krona.py=1.2}, KRONA_CLEANUP={sed=4.7}, KRONA_KTIMPORTTEXT={krona=2.8.1}, MERGE_RUNS={cat=8.30}, METAPHLAN3_MERGEMETAPHLANTABLES={metaphlan3=3.0.12}, METAPHLAN3_METAPHLAN3={metaphlan3=3.0.12}, MINIMAP2_ALIGN={minimap2=2.24-r1122}, MINIMAP2_INDEX={minimap2=2.24-r1122}, PORECHOP_PORECHOP={porechop=0.2.4}, SAMPLESHEET_CHECK={python=3.8.3}, SAMTOOLS_FASTQ={samtools=1.16.1}, SAMTOOLS_INDEX={samtools=1.16.1}, SAMTOOLS_STATS={samtools=1.16.1}, SAMTOOLS_VIEW={samtools=1.16.1}, UNTAR={untar=1.30}, Workflow={nf-core/taxprofiler=1.1.0dev}}"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "porechop": {
        "content": [
            "ERR3201952_ERR3201952.log.filtered:md5,92a485cb081b6e40e1bc3360e73f69a9"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "taxpasta": {
        "content": [
            "bracken_db1.tsv:md5,3b34e55c670b94b6382ea45dad46c0a4",
            "centrifuge_db3.tsv:md5,e863ece410193be03940cd6e106071a3",
            "diamond_db1.tsv:md5,955ebd2d69aa8f19c04873c41e58aeaf",
            "kaiju_db5.tsv:md5,1b1f70c369e3decc7947de9be4164aa9",
            "kraken2_db1.tsv:md5,881953e5a67956a2323950d5e3f4af4d",
            "kraken2_db2.tsv:md5,1ebecba3015143b8905168f72d88a4ff",
            "metaphlan_db4.tsv:md5,2e19ca1dd3904d18a53637d173ff9017"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "adapterremoval": {
        "content": [
            "2612_ERR5766176.settings:md5,e9f2de44ca1711aa609d0fd943f4b29f",
            "2612_ERR5766176_B.settings:md5,e9f2de44ca1711aa609d0fd943f4b29f",
            "2612_ERR5766180.settings:md5,1cc7909f9f70d185850780ae9a116803",
            "2613_ERR5766181.settings:md5,e3ec2f6c633a43bca4470aa7bd19a156"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "kaiju": {
        "content": [
            "kaiju_db5_combined_reports.txt:md5,6a5d8aa9a8c8507e48df14ce22664c76"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "bracken": {
        "content": [
            "bracken_db1_combined_reports.txt:md5,e31134c567abcffa7b2c4b19f73d5dea"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "diamond": {
        "content": [
            "2611_se_db1.diamond.tsv:md5,4d3baec82aeed53c6fa8803171edd212",
            "2612_se_db1.diamond.tsv:md5,f71d9b71b3f3c9261c6fb40383964013",
            "2613_se_db1.diamond.tsv:md5,575f69334dbd4d49b05ae74058d80f35",
            "ERR3201952_se_db1.diamond.tsv:md5,3c3d1a901abe0a097469423d217c13db"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "kraken2": {
        "content": [
            "kraken2_db1_combined_reports.txt:md5,01483439132d2415c10727b22df5c8cd",
            "kraken2_db2_combined_reports.txt:md5,38f7adbd186ceb2e2223c76a2fb2d591"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "centrifuge": {
        "content": [
            "centrifuge_db3_combined_reports.txt:md5,5d64a6ba245b269d2c501d614ac302f6"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "multiqc": {
        "content": [
            "bbduk.txt:md5,1d4199d37e34cf441a56bf5de8943c8f",
            "diamond.txt:md5,e036b32e0f8300062c0a976ec41e1fa5",
            "filtlong.txt:md5,69cd258c3883ba2acfb1b72cf3b743de",
            "multiqc_bowtie2.txt:md5,ecf617042c70489a52dd96b489235185",
            "multiqc_adapter_removal.txt:md5,cf7514f23cad52b24aa9f49de69bb10b",
            "multiqc_fastqc_1.txt:md5,bb3a644b4fd50572dde2791cde4f574f",
            "multiqc_fastqc.txt:md5,4e145ddc71fb1b37b62df907fa452200",
            "multiqc_general_stats.txt:md5,c8ed439ae1104e85e95a8d72119c86a6",
            "multiqc_kaiju_species.txt:md5,3fcbbc81fc9cf95b612af67f61403357",
            "multiqc_samtools_stats.txt:md5,2f6d6068b16b31a3cf9c6ea7be27c815",
            "porechop.txt:md5,5cadfceb7d9bc6b4da7bbb981a90d8ed"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "bowtie2/align/": {
        "content": [
            "2612_ERR5766176.bowtie2.log:md5,3db39ae6aad6836229e33cdf9ffd5217",
            "2612_ERR5766176_B.bowtie2.log:md5,42bbd14b66a5479a784bb7c9baad7e2e",
            "2612_ERR5766180.bowtie2.log:md5,144b4ac6ebba1972470eda16a281ba8d",
            "2613_ERR5766181.bowtie2.log:md5,fb254a4d2de3f9a5b80dfd38762c6d6e"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    },
    "samtools/stats/": {
        "content": [
            "2612_ERR5766176.stats:md5,856c7d9d15b3c3d1cbd187f71b90b489",
            "2612_ERR5766176_B.stats:md5,cf24b42b433f0a38cac7e2dc0f91dc6e",
            "2612_ERR5766180.stats:md5,1181799d32a1b33a34d8f4e036d9e512",
            "2613_ERR5766181.stats:md5,7975c55ca5dd0a841a4deeaccd19511f",
            "ERR3201952_ERR3201952.stats:md5,06a195adabd47f0d5eec9b15558a47cd"
        ],
        "timestamp": "2023-06-20T03:57:33+0000"
    }
}